src/modules/date.cpp

```diff
@@ -42,22 +42,26 @@ namespace modules {
     auto time = std::time(nullptr);
 
     auto date_format = m_toggled ? m_dateformat_alt : m_dateformat;
-    char date_buffer[64]{'\0'};
-    strftime(date_buffer, sizeof(date_buffer), date_format.c_str(), localtime(&time));
+    // Clear stream contents
+    datetime_stream.str("");
+    datetime_stream << std::put_time(localtime(&time), date_format.c_str());
+    auto date_string = datetime_stream.str();
 
     auto time_format = m_toggled ? m_timeformat_alt : m_timeformat;
-    char time_buffer[64]{'\0'};
-    strftime(time_buffer, sizeof(time_buffer), time_format.c_str(), localtime(&time));
+    // Clear stream contents
+    datetime_stream.str("");
+    datetime_stream << std::put_time(localtime(&time), time_format.c_str());
+    auto time_string = datetime_stream.str();
 
-    bool date_changed{strncmp(date_buffer, m_date.c_str(), sizeof(date_buffer)) != 0};
-    bool time_changed{strncmp(time_buffer, m_time.c_str(), sizeof(time_buffer)) != 0};
+    bool date_changed{strncmp(date_string.c_str(), m_date.c_str(), sizeof(date_string)) != 0};
+    bool time_changed{strncmp(time_string.c_str(), m_time.c_str(), sizeof(time_string)) != 0};
```

Comment:
I think that since we are using strings and not char arrays anymore, we can use the `!=` operator. `sizeof` won't work here because it isn't a char array anyways.